packages/salesforcedx-vscode-core/src/notifications/progressNotification.ts

```diff
@@ -10,20 +10,21 @@ import { nls } from '../../src/messages';
 import { CommandExecution } from '@salesforce/salesforcedx-utils-vscode/out/src/cli';
 
 export class ProgressNotification {
-  public static async show(
+  public static show(
     execution: CommandExecution,
     token: vscode.CancellationTokenSource
   ) {
-    await vscode.window.withProgress(
+    return vscode.window.withProgress(
```

Comment:
In my original pr (https://github.com/forcedotcom/salesforcedx-vscode/pull/527), I made the assumption that because `withProgress` returns a promise, I had to await the resolution of the promise before continuing on with the execution of the remaining command code. However, this was a wrong assumption. `WithProgress` sets up listeners that listen for the success/error exit codes of commands before resolving,  and for a few commands, waiting on `withProgress` to resolve actually prevented the command from even being executed at all. As a result, the listeners on `withProgress` never received exit subjects, `withProgress` never resolved, the command code was blocked from being executed, and the looping continued.